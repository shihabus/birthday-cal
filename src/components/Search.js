import React, { useState } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import get from "lodash/get";

const FormWrapper = styled.form`
  grid-area: search;
  padding-left: 5vh;
`;

const Label = styled.label`
  color: #7f8b95;
  font-size: 18px;
  display: block;
`;

const YearInput = styled.input`
  font-size: 18px;
  padding: 10px 15px;
  color: #6a6a6a;
  display: block;
  margin: 5px 0px 10px 0px;
  border-radius: 5px;
  border: none;
  &::placeholder {
    color: #a1a1a1;
  }
`;

const Submit = styled.button`
  background-color: #2889ec;
  align-self: flex-end;
  color: white;
  padding: 10px 25px;
  border-radius: 5px;
  font-size: 16px;
  border: none;
  margin: 10px 5px;
  cursor: pointer;
  &:disabled {
    cursor: no-drop;
    background-color: #b2b2b2;
    color: #e1e1e1;
  }
`;

function Input(props) {
  const { onSubmitCallBack } = props;
  const [value, setValue] = useState("");

  return (
    <FormWrapper
      aria-label="form"
      onSubmit={e => {
        e.persist();
        e.preventDefault();
        e.stopPropagation();
        onSubmitCallBack(get(e, "target[0].value", ""));
      }}
    >
      <Label htmlFor="year-input">Year</Label>
      <YearInput
        maxLength={4}
        id="year-input"
        placeholder="Search by year"
        value={value}
        onChange={e => {
          e.preventDefault();
          e.stopPropagation();
          let input = e.target.value;
          if (/^\d*\.?\d*$/.test(input)) setValue(input);
        }}
      />
      <Submit type="submit" disabled={value.length < 4} aria-label="submit">
        Update
      </Submit>
    </FormWrapper>
  );
}

export default Input;

Input.propTypes = {
  onSubmitCallBack: PropTypes.func.isRequired
};
