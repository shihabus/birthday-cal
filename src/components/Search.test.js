import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Search from "./Search";

test("Search snapshot", () => {
  const mockSubmit = jest.fn();
  const { container, getByLabelText } = render(
    <Search onSubmitCallBack={mockSubmit} />
  );
  const form = getByLabelText("form");

  fireEvent.submit(form);

  expect(mockSubmit).toHaveBeenCalledTimes(1);

  expect(container.firstChild).toMatchSnapshot();
});
