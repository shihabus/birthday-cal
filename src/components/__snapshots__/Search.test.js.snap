// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot 1`] = `
<form
  aria-label="form"
  class="sc-AxjAm fnuDkL"
>
  <label
    class="sc-AxirZ gxIIuX"
    for="year-input"
  >
    Year
  </label>
  <input
    class="sc-AxiKw eqAHUx"
    id="year-input"
    maxlength="4"
    placeholder="Search by year"
    value=""
  />
  <button
    aria-label="submit"
    class="sc-AxhCb fNtaJn"
    disabled=""
    type="submit"
  >
    Update
  </button>
</form>
`;
